ppd/ppd-test.c

```diff
@@ -0,0 +1,4036 @@
+/*
+ * PPD test program for CUPS.
+ *
+ * Copyright © 2007-2018 by Apple Inc.
+ * Copyright © 1997-2007 by Easy Software Products, all rights reserved.
+ *
+ * Licensed under Apache License v2.0.  See the file "LICENSE" for more
+ * information.
+ *
+ * PostScript is a trademark of Adobe Systems, Inc.
+ */
+
+/*
+ * Include necessary headers...
+ */
+
+#include <cups/cups-private.h>
+#include <cups/dir.h>
+#include <cups/ppd-private.h>
```

Comment:
Also remove this line, and check where functions with names starting with `_ppd` are used. check through libppd whether I have overtaken them. Using "private" functions of libppd is no problem, by the way, as you are in libppd here.

Reply:
_cupsLangPuts and _cupsLangPrintf is in langprint.c in cups, an alternative of which is not present in libppd. Should I migrate langprint.c to libppd as well?